### PR TITLE
docs: add Angular CLI autocompletion doc

### PR DESCRIPTION
**Pull request by dgp1130**
Spin off from https://github.com/angular/angular/pull/45995.

This explains how to set up autocompletion with the Angular CLI:

1. Via the automated prompt on first execution.
2. Via `ng completion`.
3. Manually via appending `source <(ng completion script)` to the user's `~/.bashrc` file.

It also specifically calls out supported and unsupported platforms as well as the requirement for a global install of the Angular CLI.

The demo `/generated/images/guide/cli/completion.gif` needs to be merged to the angular/angular repository. https://github.com/angular/angular/pull/45995 will submit that image so it can be used from this doc.

@mgechev, not sure if you want anyone from the docs team to take an edit pass on this?

/cc @twerske, @alan-agius4

### CODE REVIEW

**Review by alan-agius4 — DISMISSED** [review]

**Comment by mgechev** [comment]
@dgp1130, just shared a link with Bob and will forward to our editor as well.

File: packages/angular/cli/src/commands/completion/long-description.md [hunk 1]
```diff
@@ -1,5 +1,73 @@
-To enable Bash and Zsh real-time type-ahead autocompletion, run
-`ng completion` and restart your terminal.
+Setting up autocompletion configures your terminal so pressing the `<TAB>` key mid-typing will
```
Comment (TMDavisGoogle) [inline packages/angular/cli/src/commands/completion/long-description.md:1]:
```suggestion
Setting up autocompletion configures your terminal, so pressing the `<TAB>` key mid-typing will
```

File: packages/angular/cli/src/commands/completion/long-description.md [hunk 2]
```diff
@@ -1,5 +1,73 @@
...
+to autocomplete `ng` commands.
+
+# Serve output...
+```
+
+If you already refused the prompt, it won't ask again. But you can run `ng completion` to
+do the same thing automatically.
+
+This modifies your terminal environment to load Angular CLI autocompletion, but can't update your
+current terminal session, either restart it or run `source <(ng completion script)` directly to
```
Comment (TMDavisGoogle) [inline packages/angular/cli/src/commands/completion/long-description.md:33]:
```suggestion
current terminal session. Either restart it or run `source <(ng completion script)` directly to
```

File: packages/angular/cli/src/commands/completion/long-description.md [hunk 3]
```diff
@@ -1,5 +1,73 @@
...
+enable autocompletion in your current session.
+
+Test it out by typing `ng ser<TAB>` and it should autocomplete to `ng serve`. Ambiguous arguments
+will show all possible options and their documentation such as `ng generate <TAB>`.
```
Comment (TMDavisGoogle) [inline packages/angular/cli/src/commands/completion/long-description.md:37]:
```suggestion
will show all possible options and their documentation, such as `ng generate <TAB>`.
```

File: packages/angular/cli/src/commands/completion/long-description.md [hunk 4]
```diff
@@ -1,5 +1,73 @@
...
+
+## Manual setup
+
+Some users may have highly customized terminal setups, possibly with configuration files checked
+into source control with an opinionated structure. `ng completion` only ever appends Angular's setup
+to an existing configuration file for your current shell or creates one if none exists. If you want
```
Comment (TMDavisGoogle) [inline packages/angular/cli/src/commands/completion/long-description.md:43]:
```suggestion
to an existing configuration file for your current shell, or creates one if none exists. If you want
```

File: packages/angular/cli/src/commands/completion/long-description.md [hunk 5]
```diff
@@ -1,5 +1,73 @@
...
+
+Windows does not support autocompletion in native shells, such as Cmd and Powershell. However,
+the Angular CLI supports Git Bash and
+[Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/) using Bash or Zsh.
+
+## Global install
+
+Autocompletion works by configuring your terminal to invoke the Angular CLI on startup to load the
+setup script. This means the terminal must be able to find and execute the Angular CLI, typically
+through a global install placing the binary on the user's `$PATH`. If you get
```
Comment (TMDavisGoogle) [inline packages/angular/cli/src/commands/completion/long-description.md:67]:
```suggestion
through a global install that places the binary on the user's `$PATH`. If you get
```

**Review by TMDavisGoogle — CHANGES_REQUESTED** [review]
Light edits (punctuation, rewording for clarity) made.

**Comment by dgp1130** [comment]
@TMDavisGoogle, thanks! Applied all your suggestions.

**Review by TMDavisGoogle — APPROVED** [review]
LGTM.

**Pull request merged**